mlflow/genai/optimize/adapt.py

```diff
@@ -0,0 +1,181 @@
+import uuid
+from concurrent.futures import ThreadPoolExecutor
+from typing import TYPE_CHECKING, Any, Callable
+
+import mlflow
+from mlflow.entities.model_registry import PromptVersion
+from mlflow.environment_variables import MLFLOW_GENAI_EVAL_MAX_WORKERS
+from mlflow.genai.evaluation.utils import (
+    _convert_eval_set_to_df,
+)
+from mlflow.genai.optimize.adapters import BasePromptAdapter, get_default_adapter
+from mlflow.genai.optimize.types import EvaluationResultRecord, LLMParams, PromptAdaptationResult
+from mlflow.genai.prompts import load_prompt, register_prompt
+from mlflow.genai.utils.trace_utils import convert_predict_fn
+from mlflow.utils import gorilla
+from mlflow.utils.annotations import experimental
+from mlflow.utils.autologging_utils.safety import _wrap_patch
+
+if TYPE_CHECKING:
+    from mlflow.genai.evaluation.utils import EvaluationDatasetTypes
+
+
+@experimental(version="3.5.0")
+def adapt_prompts(
+    predict_fn: Callable[..., Any],
```

Comment:
should we call this agent or program? or if `predict_fn` is a standard in mlflow?

Reply:
`predict_fn` is the term used everywhere in mlflow, so I just wanted to stick to it.